thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/server/thrift/ThriftDescriptiveTypeInfoProvider.java

```diff
@@ -164,6 +169,21 @@ private static FieldRequirement convertRequirement(byte value) {
         }
     }
 
+    static TypeSignature toTypeSignature(Class<?> parentType, FieldMetaData fieldMetaData,
+                                         FieldValueMetaData fieldValueMetaData) {
+        if (!fieldValueMetaData.isTypedef() && !hasTypeDef(fieldValueMetaData)) {
+            return toTypeSignature(fieldValueMetaData);
+        }
+
+        try {
+            final Field field = parentType.getField(fieldMetaData.fieldName);
+            return toTypeSignature(field.getGenericType());
+        } catch (NoSuchFieldException e) {
+            // Ignore exception.
+        }
+        return TypeSignature.ofUnresolved(firstNonNull(fieldValueMetaData.getTypedefName(), "unknown"));
```

Comment:
Question) Is it possible to just use java reflection and clean up the old logic?
I ask because if a typedef is somewhere nested inside the struct, we will use generics instead of thrift objects to traverse the object tree anyways

```suggestion
        try {
            final Field field = parentType.getField(fieldMetaData.fieldName);
            return toTypeSignature(field.getGenericType());
        } catch (NoSuchFieldException e) {
            // Ignore exception.
        }
        return TypeSignature.ofUnresolved(firstNonNull(fieldValueMetaData.getTypedefName(), "unknown"));
```

Reply:
> if a typedef is somewhere nested inside the struct, we will use generics instead of thrift objects to traverse the object tree anyways

Yes, exactly. That's why I added `hasTypeDef` method.

I left the old logic because there are not so many people who use type def with Armeria(type def doesn't do anything with Java).
In that case, I thought using explicit metadata instead of using implicit reflection is better.
But not so strong on this because we can integrate the logic into one if we just use reflection.
So please let me know if you think it's better to use just reflection for all cases.

Reply:
My original concern was that if we have to troubleshoot anything, we would have to guess which code path was used.
Having said this, I'm also fine to go ahead with the fix and clean up later if necessary.

Reply:
> type def doesn't do anything with Java

I think I've seen some usage with typedef to abstract logic within the thrift idl (especially if the idl becomes very long and complex)

Reply:
I got your concern. If so, what if I extract them as a method and add documentation for that? It will make it clear?

Reply:
> I think I've seen some usage with typedef to abstract logic within the thrift idl

Oops, I didn't know about it. 😓 
So are they using DocService with the type definition?

Reply:
> We may improve DocService to display both type information and alias properly somehow.

Indeed. 😊